Sources/Theme.php

```diff
@@ -1295,129 +1295,36 @@ public static function template_header(): void
 
 		header('Content-Type: ' . $content_type . '; charset=' . (empty(Utils::$context['character_set']) ? 'ISO-8859-1' : Utils::$context['character_set']));
 
-		// We need to splice this in after the body layer.
+		// Collect layers to be added
+		$layers = [];
+
+		// Add maintenance warning if in maintenance mode and user is admin
 		if (Utils::$context['in_maintenance'] && User::$me->is_admin) {
+			$layers[] = 'maint_warning';
+		}
+
+		// Add security warning if security issues are detected
+		Utils::$context['warnings'] = Security::checkSecurityFiles();
+		if (Utils::$context['warnings']) {
+			$layers[] = 'security_warning';
+		}
+
+		// Inform user if they are banned from posting
+		if (isset($_SESSION['ban']['cannot_post'])) {
+			$layers[] = 'banned_warning';
+		}
+
+		// Insert all collected layers after the 'body' layer
+		if ($layers != []) {
 			$position = array_search('body', Utils::$context['template_layers']);
 
 			if ($position !== false) {
-				array_splice(Utils::$context['template_layers'], $position + 1, 0, ['maint_warning']);
+				array_splice(Utils::$context['template_layers'], $position + 1, 0, $layers);
```

Comment:
Would this code be better suited  for `Theme:::loadTemplatesAndLangFiles()`?

Reply:
It would fit better there in an abstract, conceptual sense, yes. But then it would be called during `Theme::load()`, which would be problematic for a number of reasons.